### PR TITLE
LOOP-1706, LOOP-1707: Checks trusted time source to identify if device time has changed

### PR DESCRIPTION
**Pull request by rickpasetto**
Checks the device's time (as reported using `Date()`) against a "trusted" NTP source (using a 3rd-party library, [TrueTime](https://github.com/instacart/TrueTime.swift)).  If it is more than 2 minutes (120 seconds) off, issue an alert.  
This check is performed either (1) when the system issues a `UIApplication.significantTimeChangeNotification` or (2) any time `loop()` is run (i.e. the `.LoopRunning` notification).  
The alert will be issued only if it hasn't been issued within the last 30 minutes (assuming the time discrepancy still exists).

[LOOP-1706](https://tidepool.atlassian.net/browse/LOOP-1706)
[LOOP-1707](https://tidepool.atlassian.net/browse/LOOP-1707)

### CODE REVIEW

File: Loop/Managers/TrustedTimeChecker.swift [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+                self.log.error("applicationSignificantTimeChange: Error getting NTP time: %@", error.localizedDescription)
+            }
+        }
+    }
+
+    private func issueTimeChangedAlert() {
+        let alertIdentifier = Alert.Identifier(managerIdentifier: "Loop", alertIdentifier: "significantTimeChange")
+        let alertTitle = NSLocalizedString("Time Change Detected", comment: "Time change alert title")
+        // TODO: remove Tidepool-isms
+        let alertBody = NSLocalizedString("Your phone’s time has been changed. Tidepool Loop needs accurate time records to make predictions about your glucose and adjust your insulin accordingly.\n\nCheck in your iPhone Settings (General / Date & Time) and verify that Set Automatically is enabled. Failure to resolve could lead to serious under-delivery or over-delivery of insulin. If you did not change the time, contact Tidepool Support.", comment: "Time change alert body")
```
Comment (ps2) [inline Loop/Managers/TrustedTimeChecker.swift:71]: This should interpolate using the bundle app name for app name, and the "contact Tidepool Support" needs to be turned into a non Tidepool specific version.  I would recommend just removing the whole sentence "If you did not change the time, contact Tidepool Support." We'll have to add it back with the Tidepool specific localization work.

**Review by ps2 — APPROVED** [review]
LGTM! I think the string needs to be updated for DIY, but I can also do that when it lands in DIY.

File: Loop/Managers/TrustedTimeChecker.swift (hunk 1 above)
Comment (rickpasetto) [reply Loop/Managers/TrustedTimeChecker.swift:71]: Yes, this is what I brought up in standup.  I am going to let this go until LOOP-1893 is addressed.

Comment (ps2) [reply Loop/Managers/TrustedTimeChecker.swift:71]: The string that is stored here is not fixed by LOOP-1893.  LOOP-1893 allows us have Tidepool Loop display Tidepool specific strings, when non-Tidepool specific strings are in the code. Right now, pre LOOP-1893, the current state is that we need to store non-Tidepool specific strings, and display non-Tidepool specific strings, so we can continue to do DIY syncs. Does that make sense? The app name interpolation is also not blocked by LOOP-1893.

**Pull request merged**